File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = []
    a = 0
    b = 0
    for i, j in zip(progresses, speeds):
        if (100-int(i)) % j == 0:
            answer.append((100-int(i)) // j)
        else:
            answer.append(((100-int(i)) // j) + 1)
    for k in answer:
        if a < k:
            day.append(b)
            b = 0
            a = k
            b += 1
        else:
            b += 1
    day.append(b)     
    return day[1:] 
        

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] day = new int[progresses.length];
        
        for (int i = 0; i < progresses.length; i++){
            if((100-progresses[i])%speeds[i] != 0){
                day[i]=(100-progresses[i])/speeds[i] + 1;
            }else{day[i]=(100-progresses[i])/speeds[i];}
        }
        List<Integer> temp = new ArrayList<>();
        int check = day[0];
        int count = 1;
        
        for (int i = 1; i < day.length; i++){
            if(check < day[i]) {
                temp.add(count);
                check = day[i];
                count = 1;
            }else{
                count ++;
            }
        } 
        temp.add(count);
        
        int[] answer = new int[temp.size()];
        for (int i = 0; i < temp.size(); i++) {
            answer[i] = temp.get(i);
        }
        
        return answer;
    }
}